### PR TITLE
fix(stdune): make sure Path.drop_prefix drops path prefixes only

### PR DESCRIPTION
**Pull request by Alizter**
Fixes #8870

### CODE REVIEW

**Comment by emillon** [comment]
FYI the issue is affecting our own test suite since in the describe tests, when replacing `lib/ocaml` to `FINDLIB`, we're replacing `lib/ocaml-compiler-libs` by `FINDLIB/-compiler-libs`. This happens with OCaml 5.1 which might have a slightly different dir layout.

**Comment by Alizter** [comment]
@rgrinberg Why was this marked as draft?

**Comment by rgrinberg** [comment]
b/c the tests are failing

**Comment by Alizter** [comment]
Sorry @rgrinberg you are correct that the failures are relevant. I had a look and I couldn't work out how to fix melange here. I might have a look again at a later date.

**Comment by Alizter** [comment]
@rgrinberg @anmonteiro Melange is now fixed. I was missing the case where we drop the entire path which should give just `Local.root`. I've updated the tests to check this case too.

**Comment by Alizter** [comment]
@rgrinberg I've created https://github.com/ocaml/dune/pull/9363 for the extra tests I added here.

File: otherlibs/stdune/test/path_tests.ml [hunk 1]
```diff
@@ -477,20 +477,34 @@ let%expect_test "drop external prefix" =
...
   Path.External.drop_prefix
     ~prefix:(Path.External.of_filename_relative_to_initial_cwd "foo/bar")
     (Path.External.of_filename_relative_to_initial_cwd "foo/barbaz")
   |> Dyn.option Path.Local.to_dyn
   |> print_dyn;
-  [%expect {| Some "baz" |}]
+  [%expect {| None |}]
+;;
+
+let%expect_test "drop entire path" =
```
[the remark this answers is not in the document]
Comment (rgrinberg) [reply otherlibs/stdune/test/path_tests.ml:496]: Tests are still being added in this PR

[the remark this answers is not in the document]
Comment (Alizter) [reply otherlibs/stdune/test/path_tests.ml:496]: Done.

**Comment by Alizter** [comment]
@rgrinberg Sorry about those extra tests, not sure what happened with git but it resolved the conflict in a funny way causing the tests to be duplicated. Should all be fixed now.

**Pull request merged**